dm/worker/subtask.go

```diff
@@ -461,16 +462,13 @@ func (st *SubTask) Result() *pb.ProcessResult {
 // Close stops the sub task.
 func (st *SubTask) Close() {
 	st.l.Info("closing")
-	if st.Stage() == pb.Stage_Stopped {
+	if !st.setStageIfNotIn([]pb.Stage{pb.Stage_Stopped, pb.Stage_Stopping}, pb.Stage_Stopping) {
 		st.l.Info("subTask is already closed, no need to close")
 		return
 	}
 
-	st.cancel()
 	st.closeUnits() // close all un-closed units
-	st.wg.Wait()
```

Comment:
please tell me that why here no neee to wait wg ? this is not same with your PR description

> the last case subtask.Close will call Unit.Close after subtask.fetchResult returns, so all three cases are not concurrent with Pause

Reply:
`subtask.fetchResult` exit -> `st.resultWg.Wait()` -> `Unit.Close()`

Reply:
i see, 
`subtask.Close()` -> `closeUnits` ->  `cancel ctx`-> `unit ctx cancel and pr chan return` ->`st.resultWg.Wait()` -> `Unit.Close()`